pandas/core/groupby/generic.py

```diff
@@ -556,8 +556,9 @@ def _transform_general(
             if common_dtype is result.dtype:
                 result = maybe_downcast_numeric(result, self._selected_obj.dtype)
 
-        result.name = self._selected_obj.name
-        result.index = self._selected_obj.index
+        obj = self._selected_obj.dropna() if self.dropna else self._selected_obj
```

Comment:
Do we need to actually drop the data itself? I thought dropna should just include whether we include the NA group (which the factorize call handled)?

Reply:
Yes you're right. We actually don't really need the whole `obj` here, just the index with NA dropped

Reply:
Actually it's fine to delete this* - so definitely not needed

* modulo a test with empty dataframe input in `pandas/tests/groupby/test_grouping.py`. But that only worked for `transform` and not other agg functions so maybe it's ok to delete the test?

Reply:
Looks like there was a desire for `agg/transform/apply` to return the same result for an empty frame: https://github.com/pandas-dev/pandas/issues/26208#issuecomment-486723454

Reply:
Ok! Deleting was maybe heavy-handed. Actually `transform` was the odd one out before this change, it's an empty index now vs an empty `RangeIndex` before. Restored and updated the test